test/test_processes.py

```diff
@@ -143,6 +143,23 @@ def define(cls, spec):
         self.assertIn('sub', process.inputs.namespace)
         self.assertEqual(process.inputs.namespace.sub, True)
 
+    def test_raise_in_define(self):
+        """Process which raises in its 'define' method. Check that the spec is not set."""
+
+        class BrokenProcess(Process):
+            @classmethod
+            def define(cls, spec):
+                super(BrokenProcess, cls).define(spec)
+                raise ValueError
+
+        with self.assertRaises(ValueError):
+            BrokenProcess.spec()
+        # Check that the error is still raised when calling .spec()
+        # a second time.
+        with self.assertRaises(ValueError):
+            BrokenProcess.spec()
+        self.assertFalse(hasattr(BrokenProcess, '_spec'))
```

Comment:
Could you remove this check?  I'm not a big fan of unit test checking implementation details.

Reply:
Done!

Reply:
Thanks!  Now just waiting for PRs on `topika` and `kiwipy` so I can release new versions and fix these tests.

Reply:
It's PRs all the way down 😄